lib/datapackage/resource.rb

```diff
@@ -6,6 +6,30 @@ class Resource < Hash
 
     attr_reader :errors, :profile, :name, :source
 
+    def self.infer(filepath)
```

Comment:
Just realized I only have a unit test for `Package#infer` and should add one for this method in `Resource` too.